Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -67,7 +67,6 @@ namespace AZ
             m_materialPipelineData = {};
             m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
             ShaderReloadNotificationBus::MultiHandler::BusDisconnect();
-            Data::AssetBus::Handler::BusDisconnect();
             if (!m_materialAsset.IsReady())
```

Comment:
This should still disconnect from the asset bus after checking for the material asset to be ready in case it was previously connected and the material was reinitialized.

Reply:
This is the Init call which is private. It is called only once when the Material is created or upon OnAssetReady(), which already disconnects. I also added BusDisconnect in the destructor in case it got destoryeed before OnAssetReady() is called. Should be safe as is. @invertednormal made me realize that that call is dedundant, so I removed it.

Reply:
Init is still called from the shader reload notification functions in this file. Need to disconnect after this check in case Init is invoked from another code path.